utils/system.go

```diff
@@ -0,0 +1,117 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/longhorn/go-common-libs/types"
+)
+
+// GetKernelRelease returns the kernel release string.
+func GetKernelRelease() (string, error) {
+	utsname := &syscall.Utsname{}
+	if err := syscall.Uname(utsname); err != nil {
+		logrus.WithError(err).Warn("Failed to get kernel release")
+		return "", err
+	}
+
+	// Extract the kernel release from the Utsname structure
+	release := make([]byte, 0, len(utsname.Release))
+	for _, b := range utsname.Release {
+		if b == 0x00 {
+			logrus.Trace("Found end of kernel release string [0x00]")
+			break
+		}
+		release = append(release, byte(b))
+	}
+	return string(release), nil
+}
+
+// GetOSDistro reads the /etc/os-release file and returns the ID field.
+func GetOSDistro(osReleaseContent string) (string, error) {
+	var err error
+	defer func() {
+		err = errors.Wrapf(err, "failed to get host OS distro")
+	}()
+
+	if types.CachedOSDistro != "" {
+		logrus.Tracef("Cached OS distro: %v", types.CachedOSDistro)
+		return types.CachedOSDistro, nil
+	}
+
+	lines := strings.Split(osReleaseContent, "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "ID=") {
+			id := strings.TrimPrefix(line, "ID=")
+			id = strings.Trim(id, `"`)
+			types.CachedOSDistro = id
```

Comment:
Indeed, it should be static, but what I mean as a lib, it's better to have the returned result explicitly instead of caching it implicitly, also if want cache, you need to handle well it concurrency if needed.